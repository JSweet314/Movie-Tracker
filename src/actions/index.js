export const loadCards = (movies) => ({
  type: 'LOAD_CARDS',
  movies
});

export const addUser = (user) => ({
  type: 'ADD_USER',
  user
});
