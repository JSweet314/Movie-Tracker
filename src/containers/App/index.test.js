import React from 'react';
import { App, mapDispatchToProps, mapStateToProps } from './index';
import { shallow } from 'enzyme';
import * as actions from '../../actions';
import mockMovies from '../../mockData';

describe('App', () => {
  let wrapper;
  const mockLoadCards = jest.fn();
  const mockCaptureUser = jest.fn();
  const mockLogOutUser = jest.fn();
  const mockChangeLogStatus = jest.fn();
  
  beforeEach( () => {
    wrapper = shallow(
      <App 
        user={{}}
        loadCards={mockLoadCards}
        captureUser={mockCaptureUser}
        logOutUser={mockLogOutUser}
        changeLogStatus={mockChangeLogStatus}
      />,
      { disableLifecycleMethods: true });
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with the right params for loadCards', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    const expected = actions.loadCards(mockMovies);

    mapped.loadCards(mockMovies);

    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should call dispatch with the right params for captureUser', () => {
    const mockUser = {name: 'taco', email: 'tac0@taco', password: 'burrito'};
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    const expected = actions.captureUser(mockUser);

    mapped.captureUser(mockUser);

    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should call dispatch with the right params for captureUser', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    const expected = actions.logOutUser();

    mapped.logOutUser();

    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should call dispatch with the right params for captureUser', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    const expected = actions.logOutUser();

    mapped.logOutUser();

    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should call dispatch with the right params for captureUser', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    const expected = actions.changeLogStatus(false);

    mapped.changeLogStatus(false);

    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  describe('mapStateToProps', () => {
    it('should return an object with user and user state', () => {
      const expected = {user: {name: 'tom'}};
      expect(mapStateToProps({ user: { name: 'tom' } })).toEqual(expected);
    });
  });
});
