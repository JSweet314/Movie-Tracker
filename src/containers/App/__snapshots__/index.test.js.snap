// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    captureUser={[Function]}
    changeLogStatus={[Function]}
    loadCards={[Function]}
    logOutUser={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    Welcome to Movie Tracker
          </h1>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/"
          >
                    Home
          </NavLink>
          <div>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/forms/signup"
                    >
                              Sign Up
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/forms/login"
                    >
                              Log In
                    </NavLink>
          </div>
</header>,
        <Switch>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    path="/forms/:id"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/favorites"
                    render={[Function]}
          />
</Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="App-title"
>
              Welcome to Movie Tracker
</h1>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/"
>
              Home
</NavLink>,
            <div>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/forms/signup"
              >
                            Sign Up
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/forms/login"
              >
                            Log In
              </NavLink>
</div>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Movie Tracker",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Welcome to Movie Tracker",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Home",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/forms/signup"
>
                  Sign Up
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/forms/login"
>
                  Log In
</NavLink>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Sign Up",
                  "to": "/forms/signup",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Log In",
                  "to": "/forms/login",
                },
                "ref": null,
                "rendered": "Log In",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              path="/forms/:id"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/forms/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Welcome to Movie Tracker
            </h1>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/"
            >
                        Home
            </NavLink>
            <div>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/forms/signup"
                        >
                                    Sign Up
                        </NavLink>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/forms/login"
                        >
                                    Log In
                        </NavLink>
            </div>
</header>,
          <Switch>
            <Route
                        exact={true}
                        path="/"
                        render={[Function]}
            />
            <Route
                        path="/forms/:id"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/favorites"
                        render={[Function]}
            />
</Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="App-title"
>
                Welcome to Movie Tracker
</h1>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/"
>
                Home
</NavLink>,
              <div>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/forms/signup"
                >
                                Sign Up
                </NavLink>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/forms/login"
                >
                                Log In
                </NavLink>
</div>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to Movie Tracker",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "Welcome to Movie Tracker",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Home",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/forms/signup"
>
                    Sign Up
</NavLink>,
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/forms/login"
>
                    Log In
</NavLink>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Sign Up",
                    "to": "/forms/signup",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Log In",
                    "to": "/forms/login",
                  },
                  "ref": null,
                  "rendered": "Log In",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
/>,
              <Route
                path="/forms/:id"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/favorites"
                render={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/forms/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
